akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala

```diff
@@ -630,10 +629,19 @@ private[akka] class LocalActorRefProvider private[akka] (
   override lazy val guardian: LocalActorRef = {
     val cell = rootGuardian.underlying
     cell.reserveChild("user")
+    val dispatcher =
+      system.guardianProps match {
```

Comment:
I wonder if in that case using `map` and `getOrElse` wouldn't be as readable as the extra comments :)

Reply:
I'd still want to explicitly point out the why with a comment, which is user code not running on internal dispatcher. I can make that a single comment above though.